### PR TITLE
observations/filter: react to zone changing

### LINKED ISSUE #391: Switching Zone Using Drop-Down Doesn't Impact Observations Filter
When you have the avalanche forecast, weather forecast or conditions blog tabs open and switch the zone using the drop-down at the top of the screen, the tab re-renders with the appropriate data. However, when the observations tab is open and the zone is changed using the drop-down, the filters set is not updated to the new zone.

https://github.com/NWACus/avy/assets/7328370/0780a993-ec7d-4a93-87b4-7c99bc8a3496

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Fixes #391

### CODE REVIEW

File: components/observations/ObservationsListView.tsx [hunk 1]
```diff
@@ -49,17 +50,25 @@ interface ObservationFragmentWithPageIndexAndZoneAndSource extends ObservationFr
...
+    [initialFilterConfig],
+  );
   const [filterConfig, setFilterConfig] = React.useState<ObservationFilterConfig>(originalFilterConfig);
   const [filterModalVisible, setFilterModalVisible] = React.useState<boolean>(false);
   const mapResult = useMapLayer(center_id);
   const mapLayer = mapResult.data;
 
+  // when the initial filter inputs change, we should honor those and try to overlay the user's requests on top of the new base
+  React.useEffect(() => {
+    setFilterConfig(current => _.assign({}, current, originalFilterConfig));
```
Comment (floatplane) [inline components/observations/ObservationsListView.tsx:69]: I don't think you need lodash here? Also the comment is not right I think...or at least the code is not doing what the comment says. It's laying the new base on top of the user's requests, not vice-versa.

```suggestion
    setFilterConfig(current => {
      ...current,
      ...originalFilterConfig
    });
```

**Review by floatplane — COMMENTED** [review]
ptal @stevekuznetsov

**Comment by stevekuznetsov** [comment]
@floatplane updated to use destructuring

**Comment by floatplane** [comment]
looks like prettier wants a change, but LGTM!

**Review by floatplane — APPROVED** [review]

**Pull request merged**